server/server.go

```diff
@@ -79,12 +80,45 @@ var dirPaths = []string{
 	"trie",
 }
 
+// set log output to the designated file
+func setLogFileWriter(logFilePath string) (*os.File, error) {
+	logFile, err := os.Create(logFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("could not create log file, %w", err)
+	}
+
+	// if file path is empty string return error
+	if logFilePath == "" {
+		return nil, fmt.Errorf("no file path defined")
+	}
+
+	return logFile, nil
+}
+
+// create new logger; log to file if log path is defined
+func newLogger(config *Config) hclog.Logger {
+	var logFile *os.File
+
+	// if there is an error output logs to console
+	logFile, err := setLogFileWriter(config.LogFilePath)
+	if err != nil {
+		return hclog.New(&hclog.LoggerOptions{
+			Name:  "polygon",
+			Level: config.LogLevel,
+		})
+	}
```

Comment:
I guess logger will output to console when `config.LogFilePath` is empty string. It is normal behaviour, so you need to check `config.LogFilePath` (not by error handling) and create console logger if it's empty string.

Reply:
The idea here is that we fallback to console output if `config.LogFilePath` is empty string or if there are any errors when creating the specified log file.

Added the console output regarding log file in d6412ffed8122cf3b77bdbd9b128b4bb181f226e 
When writing to console, if `--log-to` flag is not set, the first line will be `2022-04-28T11:25:49.825+0200 [INFO]  polygon: Log file: fallback to console output err="no file path defined"` or it will show an error on why the log file could not be created.